conduit/data/datasets/vision/nih.py

```diff
@@ -0,0 +1,189 @@
+from enum import Enum
+from pathlib import Path
+from typing import ClassVar, Optional, Union, cast
+
+import pandas as pd  # type: ignore
+from ranzen import parsable, str_to_enum
+from sklearn.preprocessing import MultiLabelBinarizer  # type: ignore
+import torch
+from typing_extensions import TypeAlias
+
+from conduit.data.datasets.utils import ImageTform
+from conduit.data.datasets.vision.base import CdtVisionDataset
+
+__all__ = ["NIHChestXRays"]
+
+
+class NIHSplit(Enum):
+    TRAIN = "train_val_list.txt"
+    TEST = "test_list.txt"
+
+
+class NIHSensAttr(Enum):
+    GENDER = "Patient Gender"
+    AGE = "Patient Age"
+
+
+class NIHTargetAttr(Enum):
+    """
+    Taget attributes for the NIH chest X-rays dataset.
+
+    Fraction of labels that are positive for each of the thoracic diseases:
+        Atelectasis           0.103095
+        Cardiomegaly          0.024759
+        Consolidation         0.041625
+        Edema                 0.020540
+        Effusion              0.118775
+        Emphysema             0.022440
+        Fibrosis              0.015037
+        Hernia                0.002025
+        Infiltration          0.177435
+        Mass                  0.051570
+        Nodule                0.056466
+        Pleural_Thickening    0.030191
+        Pneumonia             0.012763
+        Pneumothorax          0.04728
+    """
+
+    ATELECTASIS = "Atelectasis"
+    CARDIOMEGALY = "Cardiomegaly"
+    CONSOLIDATION = "Consolidation"
+    EDEMA = "Edema"
+    EFFUSION = "Effusion"
+    EMPHYSEMA = "Emphysema"
+    FIBROSIS = "Fibrosis"
+    HERNIA = "Hernia"
+    INFILTRATION = "Infiltration"
+    MASS = "Mass"
+    NODULE = "Nodule"
+    PLEURAL_THICKENING = "Pleural_Thickening"
+    PNEUMONIA = "Pneumonia"
+    PNEUMOTHORAX = "Pneumothorax"
+    FINDING = "No Finding"
+
+
+class NIHChestXRays(CdtVisionDataset):
+    """ "
+    National Institutes of Health (NIH) chest X-Rays dataset.
+    This NIH Chest X-rays dataset is comprised of 112,120 X-ray images with disease labels from
+    30,805 unique patients. To create these labels, the authors used Natural Language Processing to
+    text-mine disease classifications from the associated radiological reports. The labels are
+    expected to be >90% accurate and suitable for weakly-supervised learning. The original radiology
+    reports are not publicly available but you can find more details on the labeling process in
+    `this
+    <https://www.nih.gov/news-events/news-releases/nih-clinical-center-provides-one-largest-publicly-available-chest-x-ray-datasets-scientific-community>`__
+    Open Access paper.
+
+    The dataset can be downloaded by following the above link or from `kaggle <https://www.kaggle.com/datasets/nih-chest-xrays/data>`__
+    """
+
+    TARGET_ATTR: TypeAlias = NIHTargetAttr
+    SENS_ATTR: TypeAlias = NIHSensAttr
+    SPLIT: TypeAlias = NIHSplit
```

Comment:
I don't think the TypeAliases should be in ALL CAPS?

Reply:
👍